source/src/main/groovy/com/kezong/fataar/FlavorArtifact.groovy

```diff
@@ -152,11 +152,8 @@ class FlavorArtifact {
                 }
             }
 
-            if (variant.productFlavors.isEmpty()) {
-                return false
-            }
             // 2. find missingStrategies
-            ProductFlavor flavor = variant.productFlavors.first()
+            ProductFlavor flavor = variant.productFlavors.isEmpty() ? variant.mergedFlavor : variant.productFlavors.first()
             flavor.missingDimensionStrategies.find { entry ->
                 String toDimension = entry.getKey()
                 String toFlavor = entry.getValue().getFallbacks().first()
```

Comment:
@kezong should't we first try the requested value and then try all the fallbacks?
```suggestion
                String toFlavor = entry.getValue().requested
```
Because the semantic of missingDimensionStrategy is as follows:
`missingDimensionStrategy 'dimension', 'requested', 'fallback1', 'fallback2' ...`

`getFallbacks().first()` would select `fallback1` and skipping `requested` and ignoring the other fallbacks.

As a result `missingDimensionStrategy 'dimension', 'requested'` causes an exception, because it has not fallbacks.